core/src/main/java/org/sonarsource/sonarlint/core/container/connected/objectstore/ObjectStore.java

```diff
@@ -30,7 +31,7 @@
  */
 public interface ObjectStore<K, V> {
 
-  void write(K key, V value) throws IOException;
+  void write(K key, Iterator<V> values) throws IOException;
 
-  Optional<V> read(K key) throws IOException;
+  Optional<Iterator<V>> read(K key) throws IOException;
```

Comment:
Can't we introduce the iterator at another level of abstraction? Either in `SimpleObjectStore` or in `ServerIssueStore`?